docs/caveats.rst

```diff
@@ -16,8 +15,27 @@ all exponents as part of the protocol; then for each
 sharing. In practice this exponent should be a lower bound for any exponent that
 would naturally arise.
 
+.. _alternative-base:
+
+Alternative Base for EncodedNumber
+----------------------------------
+
+*If* you need to interact with a library using another base, create a simple subclass
+of :class:`paillier.EncodedNumber` and ensure you include the `BASE` and `LOG2_BASE`
+attributes::
+
+    class AltEncodedNumber(paillier.EncodedNumber):
+        BASE = 2
+        LOG2_BASE = math.log(BASE, 2)
+
+
+.. warning::
+
+    Be careful when designing protocols with `BASE = 2`. If you don't require specific a
+    specific value for the unencrypted exponents after an operation then you're potentially
+    leaking a lot of information about what happened.
```

Comment:
maybe I'd phrase this:
"As always, if you don't require a specific value for the unencrypted exponents after an operation, then might be leaking information about what happened - but with small bases this problem is exacerbated."